### PR TITLE
[file_selector] Add MIME type support on macOS

### PR DESCRIPTION
**Pull request by stuartmorgan**
Adds a macOS 11+ codepath that uses `UTType`s, allowing for supporting MIME type (and moving off of the deprecated `allowedFileTypes`).

Fixes https://github.com/flutter/flutter/issues/117843

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/file_selector/file_selector_macos/macos/Classes/messages.g.swift
Comment (stuartmorgan) [inline packages/file_selector/file_selector_macos/macos/Classes/messages.g.swift:1]: The changes to this file are just my running `swift-format` on it, which I hadn't done previously; we don't yet have enforcement of formatting for Swift in CI.

File: packages/file_selector/file_selector_macos/macos/Classes/FileSelectorPlugin.swift [hunk 1]
```diff
@@ -96,16 +99,31 @@ public class FileSelectorPlugin: NSObject, FlutterPlugin, FileSelectorApi {
...
-      allowedTypes.append(contentsOf: acceptedTypes.utis.map({ $0! }))
-      // TODO: Add support for mimeTypes in macOS 11+. See
-      // https://github.com/flutter/flutter/issues/117843
-
-      if !allowedTypes.isEmpty {
-        panel.allowedFileTypes = allowedTypes
+      if #available(macOS 11, *), !forceLegacyTypes {
+        var allowedTypes: [UTType] = []
+        // The array values are non-null by convention even though Pigeon can't currently express
+        // that via the types; see messages.dart.
```
Comment (cbracken) [inline packages/file_selector/file_selector_macos/macos/Classes/FileSelectorPlugin.swift:105]: Here and below: is there a pigeon bug for that? If so, perhaps link here to make searching for these easier later on when we fix it.

**Review by cbracken — APPROVED** [review]
lgtm! One optional suggestion.

File: packages/file_selector/file_selector_macos/macos/Classes/FileSelectorPlugin.swift (hunk 1 above)
Comment (stuartmorgan) [reply packages/file_selector/file_selector_macos/macos/Classes/FileSelectorPlugin.swift]: Done.

Comment (stuartmorgan) [reply packages/file_selector/file_selector_macos/macos/Classes/FileSelectorPlugin.swift]: Done

**Pull request merged**